xarray/core/variable.py

```diff
@@ -742,7 +742,11 @@ def _getitem_with_mask(self, key, fill_value=dtypes.NA):
 
             data = as_indexable(self._data)[actual_indexer]
             mask = indexing.create_mask(indexer, self.shape, data)
-            data = duck_array_ops.where(mask, fill_value, data)
+            if isinstance(mask, bool):
+                mask = not mask
+            else:
+                mask = ~mask
```

Comment:
Could you flip the argument order rather than adding this? I’m a little puzzles here.

Reply:
If the concern here is about consistency when applying `~` to `bool` objects and boolean dtype arrays, explicitly calling `np.logical_not` is a good alternative.

But it does feel a little weird to me to see this here. Maybe changing `duck_array_ops.notnull` would have the same effect?

Reply:
same as the `fillna` issue above, in order to get the results in the units of `data`, we need to flip the arguments and for that I need invert the mask (if there is a different way to flip the arguments without inverting, please do tell me).

I tried to use `mask = ~mask`, but `~` does not work as expected for `bool`. I'll use `np.logical_not` instead.